### ISSUE TITLE
[defect]: v3.2.5 suddenly terminates when FR has received response to status check 

### ISSUE THREAD

**Issue by hgot07** [FreeRADIUS/freeradius-server#5379]
### What type of defect/bug is this?

Crash or memory corruption (segv, abort, etc...)

### How can the issue be reproduced?

FreeRADIUS 3.2.5 is configured so that DEFAULT requests are sent to another host over the legacy udp transport. The peer is running radsecproxy and doing Dynamic Peer Discovery.
The peer fails to determine the destination, and FreeRADIUS tries a status check, which fails once. Then, FreeRADIUS terminates suddenly after it has received a response to status check.
v3.2.3 was working quite well for many months. I made configuration files for v3.2.5 based on the latest source, not using the ones for 3.2.3.

### Log output from the FreeRADIUS daemon

```shell
Here's the dying message by radiusd -fxx -l stdout. (addresses masked)
XX.XX.208.29 is the radsecproxy host.

(870) Using Post-Auth-Type Challenge
(870) # Executing group from file /usr/local/freeradius/3.2.5/etc/raddb/sites-enabled/default
(870)   Challenge { ... } # empty sub-section is ignored
(870) Sent Access-Challenge Id 141 from XX.XX.174.78:1812 to XX.XX.95.72:3595
1 length 564
(870)   EAP-Message = 0x010601f11900cad885d67b425aa2d21358a97189c6fca11db3c1861f
e900f5482934dc201b9ac8783b9dbd24e7c1d863344fba3ac35a1aee41b2e3f30226ccdf8916fc0f
86499580edf59cb6d27f5979ce05ee992c248ac7b7e577185167728ebefb60582ac81974291213c4
fbd87401128848cd5d41b7c37c41b9c19dc0c8a6fa9bfa643b1384927af11402f01285e09ed6977f
4b87160303014d0c0001490300174104bbb3772cc2342d778856f284bd3d93722d61210e5ab19490
a592056d5c7e243184af94eb045e1642ed9a9d5f5332b9de0dc51c434ab25b9c5af348d089cb08e6
0401010055473f587e955c19f25dc01640016eb74692d1482c902fc2f0db7a4faf32831f576dfd8f
7e19744ecf106d02968d503855a3c8ba7d2915951faa1a39a79a33d989dfa05ebf6043209185136a
c940bfed9aacac289e09088853be80e9a36e7eef2cf80d874c06d5cc8adfaebd61f935c44524a641
2c71f3206e499c64b44f7a1b7ea3f6dae24f065a4617283dd432d0
(870)   Message-Authenticator = 0xc54790426052a1b213730d2ddb17a9f7
(870)   State = 0xc81da807cc1bb16a2c626c482b09aa16
(870)   Proxy-State = 0x31
(870)   Proxy-State = 0x3439
(870) Finished request
Thread 2 waiting to be assigned a request
(466) No proxy response, giving up on request and marking it done
Marking home server XX.XX.208.29 port 11813 as zombie (it has not responded in 30.000000 seconds).
PING: Waiting 4 seconds for response to ping
(872) Sent Status-Server Id 220 from 0.0.0.0:47421 to XX.XX.208.29:11813 length 70
(872)   Message-Authenticator := 0x00
(872)   NAS-Identifier := "Status Check 0. Are you alive?"
PING: Next status packet in 30 seconds
(466) ERROR: Failing proxied request for user "XXXXXX@XXXX.XX", due to lack of any response from home server XX.XX.208.29 port 11813
Thread 1 got semaphore
Thread 1 handling request 466, (337 handled so far)
(466) Clearing existing &reply: attributes
(466) Found Post-Proxy-Type Fail-Accounting
(466) server default {
(466)   Post-Proxy-Type sub-section not found.  Ignoring.
(466)   # Executing group from file /usr/local/freeradius/3.2.5/etc/raddb/sites-enabled/default
(466) }
(466) Not sending reply to client.
(466) Finished request
Thread 1 waiting to be assigned a request
(872) Marking home server XX.XX.208.29 port 11813 alive
(872) Received response to status check 872 ID 220 (1 in current sequence)
# <-- radiusd exists here.
```

**Comment by alandekok** [comment]
Thanks.  We've pushed fixes for this in commit 3a944953, along with one or two other issues we've seen.  We will be releasing 3.2.6 shortly in order to address these issues.

In the mean time, please try the v3.2.x branch from github.  It should correct this issue.

**Comment by hgot07** [comment]
Thanks. I'll check.
